bukkit/src/main/java/com/github/games647/fastlogin/bukkit/hook/floodgate/FloodgateHook.java

```diff
@@ -0,0 +1,89 @@
+package com.github.games647.fastlogin.bukkit.hook.floodgate;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.bukkit.Bukkit;
+import org.geysermc.floodgate.api.FloodgateApi;
+import org.geysermc.floodgate.api.player.FloodgatePlayer;
+
+import com.github.games647.craftapi.model.Profile;
+import com.github.games647.craftapi.resolver.RateLimitException;
+import com.github.games647.fastlogin.bukkit.FastLoginBukkit;
+import com.github.games647.fastlogin.core.shared.LoginSource;
+
+public class FloodgateHook {
+
+    FastLoginBukkit plugin;
+
+    public FloodgateHook(FastLoginBukkit plugin) {
+        this.plugin = plugin;
+    }
+
+    /**
+     * Check if the player's name conflict's an existing Java player's name, and
+     * kick them if it does
+     * 
+     * @param core     the FastLoginCore
+     * @param username the name of the player
+     * @param source   an instance of LoginSource
+     * @param plugin   the FastLoginBukkit plugin
+     */
+    public void checkNameConflict(String username, LoginSource source, FloodgatePlayer floodgatePlayer) {
+        String allowConflict = plugin.getCore().getConfig().get("allowFloodgateNameConflict").toString().toLowerCase();
+
+        // check if the Bedrock player is linked to a Java account 
+        boolean isLinked = floodgatePlayer.getLinkedPlayer() != null;
+
+        if (allowConflict.equals("false")
+                || allowConflict.equals("linked") && !isLinked) {
+
+            // check for conflicting Premium Java name
+            Optional<Profile> premiumUUID = Optional.empty();
+            try {
+                premiumUUID = plugin.getCore().getResolver().findProfile(username);
```

Comment:
This makes an additional UUID request. It's very rate-limited, so can't we reuse the existing check from the login request?